bpfman-operator/controllers/bpfman-agent/common.go

```diff
@@ -533,6 +589,7 @@ func ProcessMapOwnerParam(ctx context.Context,
 	}
 }
 
+// get Clientset returns a kubernetes clientset.
```

Comment:
Andrew:
We can't rely on this always being set there's a different process for generating a k8s clientset for in-cluster vs out of cluster see https://github.com/bpfman/bpfman/blob/main/bpfman-operator/pkg/helpers/helpers.go#L109

I think we can actually just reuse getk8sConfigOrDie here

Andre:
I tried using GetClientOrDie(), but wasn't able to get it working.  My code is using a *kubernetes.Clientset, but GetClientOrDie() returns a *bpfmanclientset.Clientset.  Maybe there's a way to convert between the two or use *bpfmanclientset.Clientset to get the pod list with the filters needed, but I'll need to work on it more.  The other thought is to maybe use the pattern in getk8sConfigOrDie() to handle the in-cluster vs out of cluster case but get a *kubernetes.Clientset.  I'll need to investigate more.

Reply:
Ah ok so yeah that code in helpers is for our API specifically, however we still need to follow the same logic for generating the raw K8s client i.e if we're in-cluster see https://github.com/kubernetes/client-go/tree/master/examples/in-cluster-client-configuration for a good example